packages/js/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -188,12 +202,16 @@ Please update the local dependency on "${depName}" to be a valid semantic versio
             console.log(
               `Added the dist-tag ${tag} to v${currentVersion} for registry ${registry}.\n`
             );
-          } else {
+          } else if (isDryRun) {
```

Comment:
It doesn't make sense to print this dry-run message given it doesn't corresponding to an action when not in dry-run.

It's this entire dist tag related section that should be in an 

```ts
if (isNpmInstalled) {

}
```

i.e. roughly lines 188-284 from what I can tell from this view